IdentityCore/src/webview/embeddedWebview/MSIDAADOAuthEmbeddedWebviewController.m

```diff
@@ -82,6 +82,7 @@ - (BOOL)decidePolicyAADForNavigationAction:(WKNavigationAction *)navigationActio
         decisionHandler(WKNavigationActionPolicyCancel);
         [MSIDPKeyAuthHandler handleChallenge:requestURL.absoluteString
                                      context:self.context
+                               customHeaders:self.customHeaders
                            completionHandler:^(NSURLRequest *challengeResponse, NSError *error) {
                                if (!challengeResponse)
                                {
```

Comment:
I don't think you need to create a separate property for customHeaders. You can add the custom headers that were passed in to the challengeResponse URLRequest here.
`for (NSString *headerKey in self.customHeaders)
            [challengeResponse addValue:self.customHeaders[headerKey] forHTTPHeaderField:headerKey];`

So adding all the headers that were initially passed in here. I don't think there is any harm to pass along all the headers that were initially passed in

Reply:
I am seeing the benefit for adding all customHeaders. However, there could be a benefit of adopting fine control over what needs to be included in the response header and therefore we're picking what matters to the response. I am afraid we would be hiding an issue by setting all headers first and second setting these values in different location may not be ideal since it would be confusing where the headers are set.

Reply:
Agree with @antonioalwan, this layer shouldn't know about PKeyAuth logic.